accumulator/forestdata.go

```diff
@@ -1,23 +1,55 @@
 package accumulator
 
 import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"errors"
 	"fmt"
+	"io"
+	"io/ioutil"
 	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
 )
 
 // leafSize is a [32]byte hash (sha256).
 // Length is always 32.
 const leafSize = 32
 
+// This is for error checking. Rather expensive if you have it on
+// Worth it to be on for now
+var sanity bool = true
+
 // ForestData is the thing that holds all the hashes in the forest.  Could
 // be in a file, or in ram, or maybe something else.
 type ForestData interface {
+	// returns the hash value at the given position
 	read(pos uint64) Hash
+
+	// writes the given hash at the given position
 	write(pos uint64, h Hash)
+
+	// for the given two positions, swap the hash values
 	swapHash(a, b uint64)
+
+	// given positions a and b, take the width value (w) and swap
+	// all the positions widthin it.
 	swapHashRange(a, b, w uint64)
+
+	// returns how many leaves the current forest can hold
 	size() uint64
+
+	// allocate more space to the forest. newSize should be in leaf count (bottom row of the forest)
+	// can't resize down
 	resize(newSize uint64) // make it have a new size (bigger)
+
+	// just needed for cowForest since you need the forestRows value to
+	// calculate the leave positions for treeBlock
+	setRow(row uint8)
```

Comment:
could this also be done through `resize`?

Reply:
Fixed to use resize